### PR TITLE
add SSP operator

### PR DESCRIPTION
**Pull request by fromanirh**
The SSP operator is unique because
- it is a pure ansible-operator, so we added (mostly)autogenerated
  go APIs to facilitate the integration with HCO
- it manages and wants to manage more than one CR

Signed-off-by: Francesco Romani <fromani@redhat.com>

### CODE REVIEW

**Comment by fromanirh** [comment]
some TODOs still pending

**Comment by rthallisey** [comment]
also add ssp to `deploy.sh` and `clean.sh` so it gets picked up by the Makefile

File: pkg/controller/hyperconverged/hyperconverged_controller.go [hunk 1]
```diff
@@ -30,6 +32,7 @@ var (
 	KubeVirtImagePullPolicy      = "IfNotPresent"
 	CDIImagePullPolicy           = "IfNotPresent"
 	NetworkAddonsImagePullPolicy = "IfNotPresent"
```
Comment (rthallisey) [inline pkg/controller/hyperconverged/hyperconverged_controller.go]: The vars are going to be removed, so need to add and `ImagePullPolicy`

File: pkg/controller/hyperconverged/hyperconverged_controller.go [hunk 2]
```diff
@@ -135,6 +161,12 @@ func (r *ReconcileHyperConverged) Reconcile(request reconcile.Request) (reconcil
 		NetworkAddonsImagePullPolicy = instance.Spec.NetworkAddonsImagePullPolicy
 	}
 
+	if instance.Spec.SSPImagePullPolicy != "" {
```
Comment (rthallisey) [inline pkg/controller/hyperconverged/hyperconverged_controller.go]: Also no longer need this if.

File: pkg/controller/hyperconverged/hyperconverged_controller.go (hunk 1 above)
Comment (fromanirh) [reply pkg/controller/hyperconverged/hyperconverged_controller.go]: Not sure I understood, could you please rephrase? keeping the hunk temporarily.

File: pkg/controller/hyperconverged/hyperconverged_controller.go (hunk 2 above)
Comment (fromanirh) [reply pkg/controller/hyperconverged/hyperconverged_controller.go]: Removed.

File: pkg/controller/hyperconverged/hyperconverged_controller.go (hunk 1 above)
Comment (rthallisey) [reply pkg/controller/hyperconverged/hyperconverged_controller.go]: The HCO is not going to expose any component configuration right now.  We're removing it in #40.  So remove `SSPImagePullPolicy` and any code that uses it.

Comment (fromanirh) [reply pkg/controller/hyperconverged/hyperconverged_controller.go]: ack.

**Comment by fabiand** [comment]
@tiraboschi can you pick up this PR as @fromanirh will not be able to work on this during the next week,

**Pull request merged**